# Introduction to AgentChat API with First AssistantAgent  
We’ll use the AgentChat API to create a simple `AssistantAgent` and explore its capabilities.  

In [1]:
%pip install autogen-agentchat autogen-ext openai tiktoken


Note: you may need to restart the kernel to use updated packages.


In [2]:
from autogen_agentchat.agents import AssistantAgent
from autogen_ext.models.openai import OpenAIChatCompletionClient
from autogen_core.models import ModelFamily
import os

api_key = ""  # put  your api key in " "


In [3]:
model_client = OpenAIChatCompletionClient(
    model="llama-3.3-70b-versatile",
    base_url="https://api.groq.com/openai/v1",
    api_key=api_key,
    model_info={
        "vision": False,
        "function_calling": True,
        "json_output": True,
        "structured_output": True,
        "family": ModelFamily.UNKNOWN,
    }
)


## Creating Our First AssistantAgent  
The `AssistantAgent` is a versatile agent for conversations, powered by AgentChat.  

In [4]:
assistant = AssistantAgent(name = "assistant",model_client=model_client,description='A Basic first Agent')

## Testing the AssistantAgent  
We’ll use the `run` method to send a task and get a response.  

In [5]:
result = await assistant.run(task="What's the capital of USA & write some more about the same ?")

print(result)

messages=[TextMessage(id='e64580a1-bc68-4af1-a70c-4b7a64e657bb', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2026, 6, 28, 15, 45, 37, 843079, tzinfo=datetime.timezone.utc), content="What's the capital of USA & write some more about the same ?", type='TextMessage'), TextMessage(id='d888e4a8-32c2-423a-b596-1e45d74d5383', source='assistant', models_usage=RequestUsage(prompt_tokens=73, completion_tokens=417), metadata={}, created_at=datetime.datetime(2026, 6, 28, 15, 45, 42, 81849, tzinfo=datetime.timezone.utc), content="The capital of the United States of America (USA) is Washington, D.C. (short for District of Columbia). It is a federal district, not part of any state, and serves as the permanent capital of the country.\n\nWashington, D.C. is located on the east coast of the United States, along the Potomac River, and is home to many iconic landmarks and institutions, including:\n\n1. The White House: The official residence and workplace of the President o

messages=[TextMessage(source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 6, 18, 21, 11, 31, 850662, tzinfo=datetime.timezone.utc), content="What's the capital of France ?", type='TextMessage'), TextMessage(source='assistant', models_usage=RequestUsage(prompt_tokens=42, completion_tokens=9), metadata={}, created_at=datetime.datetime(2025, 6, 18, 21, 11, 32, 966650, tzinfo=datetime.timezone.utc), content='The capital of France is Paris. TERMINATE', type='TextMessage')] stop_reason=None

In [6]:
print(result.messages)

[TextMessage(id='e64580a1-bc68-4af1-a70c-4b7a64e657bb', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2026, 6, 28, 15, 45, 37, 843079, tzinfo=datetime.timezone.utc), content="What's the capital of USA & write some more about the same ?", type='TextMessage'), TextMessage(id='d888e4a8-32c2-423a-b596-1e45d74d5383', source='assistant', models_usage=RequestUsage(prompt_tokens=73, completion_tokens=417), metadata={}, created_at=datetime.datetime(2026, 6, 28, 15, 45, 42, 81849, tzinfo=datetime.timezone.utc), content="The capital of the United States of America (USA) is Washington, D.C. (short for District of Columbia). It is a federal district, not part of any state, and serves as the permanent capital of the country.\n\nWashington, D.C. is located on the east coast of the United States, along the Potomac River, and is home to many iconic landmarks and institutions, including:\n\n1. The White House: The official residence and workplace of the President of the Uni

In [7]:
print(result.messages[-1].content)

The capital of the United States of America (USA) is Washington, D.C. (short for District of Columbia). It is a federal district, not part of any state, and serves as the permanent capital of the country.

Washington, D.C. is located on the east coast of the United States, along the Potomac River, and is home to many iconic landmarks and institutions, including:

1. The White House: The official residence and workplace of the President of the United States.
2. The United States Capitol: The seat of the US Congress, where the Senate and House of Representatives meet.
3. The Supreme Court: The highest court in the land, which interprets the laws and the Constitution.
4. The Smithsonian Institution: A group of museums and research centers that showcase American history, art, science, and culture.
5. The National Mall: A large public park that stretches from the Lincoln Memorial to the United States Capitol, and is home to many monuments, memorials, and museums.

Washington, D.C. is a city

In [8]:
result = await assistant.run(task="What's the capital of USA & write some more about the same ?")

print(result)

messages=[TextMessage(id='ec1718ce-b345-49bf-a834-d99f07730aed', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2026, 6, 28, 15, 45, 42, 100924, tzinfo=datetime.timezone.utc), content="What's the capital of USA & write some more about the same ?", type='TextMessage'), TextMessage(id='fd28b6f7-c1d2-49cf-8a45-ced5ebf0e6f1', source='assistant', models_usage=RequestUsage(prompt_tokens=513, completion_tokens=441), metadata={}, created_at=datetime.datetime(2026, 6, 28, 15, 45, 43, 531656, tzinfo=datetime.timezone.utc), content="The capital of the United States of America (USA) is Washington, D.C. (short for District of Columbia). It is a federal district, not part of any state, and serves as the permanent capital of the country.\n\nWashington, D.C. is a city with a rich history, dating back to 1790, when it was founded as the capital of the United States. The city has a diverse population of over 700,000 people, and is a hub for politics, culture, education, and 

messages=[TextMessage(source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 6, 18, 21, 14, 38, 430090, tzinfo=datetime.timezone.utc), content="What's the capital of USA & write some more about the same ?", type='TextMessage'), TextMessage(source='assistant', models_usage=RequestUsage(prompt_tokens=73, completion_tokens=393), metadata={}, created_at=datetime.datetime(2025, 6, 18, 21, 14, 45, 842318, tzinfo=datetime.timezone.utc), content="The capital of the United States is Washington, D.C.\n\n### Washington, D.C.\n- **Location:** Washington, D.C. is located on the east coast of the United States along the Potomac River, bordering the states of Virginia and Maryland.\n  \n- **History:** The Residence Act of 1790 approved the creation of a capital district located along the Potomac River. Washington, D.C. was named after George Washington, the first President of the United States, and Christopher Columbus.\n\n- **Government:** As the nation's capital, Washington, D.C. is the seat of the U.S. federal government. It houses the three branches of the federal government:\n  - The **Executive Branch**, led by the President, who resides in the White House.\n  - The **Legislative Branch**, with the two houses of Congress— the Senate and the House of Representatives— meeting at the U.S. Capitol.\n  - The **Judicial Branch**, with the Supreme Court of the United States, the highest court in the country.\n\n- **Landmarks:** Washington, D.C. is well-known for its historic landmarks and monuments, including the Washington Monument, Lincoln Memorial, Jefferson Memorial, and the United States Capitol. It is also home to numerous museums, particularly those of the Smithsonian Institution.\n\n- **Economy and Culture:** The city's economy is driven largely by the federal government, but it also has a growing tourism industry, with millions of visitors each year. Additionally, Washington, D.C. has a vibrant cultural scene characterized by theaters, music venues, and a diverse population.\n\n- **Education:** The city is home to several major universities, including Georgetown University, George Washington University, and American University.\n\nDespite being the capital, Washington, D.C. is not part of any state and operates as a separate federal district. Residents of D.C. have a unique political status, with limited self-governance and representation in Congress. TERMINATE", type='TextMessage')] stop_reason=None

In [9]:
print(result.messages[-1].content)

The capital of the United States of America (USA) is Washington, D.C. (short for District of Columbia). It is a federal district, not part of any state, and serves as the permanent capital of the country.

Washington, D.C. is a city with a rich history, dating back to 1790, when it was founded as the capital of the United States. The city has a diverse population of over 700,000 people, and is a hub for politics, culture, education, and tourism.

Some of the key attractions in Washington, D.C. include:

1. The White House: The official residence and workplace of the President of the United States.
2. The United States Capitol: The seat of the US Congress, where the Senate and House of Representatives meet.
3. The National Mall: A large public park that stretches from the Lincoln Memorial to the United States Capitol, and is home to many monuments, memorials, and museums.
4. The Smithsonian Institution: A group of museums and research centers that showcase American history, art, science